Index: cifs-utils-6.0/mount.cifs.c
===================================================================
--- cifs-utils-6.0.orig/mount.cifs.c	2013-03-25 11:52:05.000000000 -0300
+++ cifs-utils-6.0/mount.cifs.c	2013-06-27 20:31:56.144316008 -0300
@@ -1627,12 +1627,13 @@
 {
 #ifdef ENABLE_SYSTEMD
 	int is_systemd_running;
-	struct stat a, b;
+	struct stat a, b, c;
 
 	/* We simply test whether the systemd cgroup hierarchy is
 	 * mounted */
 	is_systemd_running = (lstat("/sys/fs/cgroup", &a) == 0)
 		&& (lstat("/sys/fs/cgroup/systemd", &b) == 0)
+		&& (lstat("/bin/systemd-ask-password", &c) == 0)
 		&& (a.st_dev != b.st_dev);
 
 	if (is_systemd_running) {
